src/titiler/xarray/titiler/xarray/factory.py

```diff
@@ -0,0 +1,364 @@
+            ] = None,
+            multiscale: Annotated[
+                Optional[bool],
+                Query(
+                    title="multiscale",
+                    description="Whether the dataset has multiscale groups (Zoom levels)",
+                ),
+            ] = None,
+            src_path=Depends(self.path_dependency),
+            reader_params=Depends(self.reader_dependency),
+            tile_params=Depends(self.tile_dependency),
+            layer_params=Depends(self.layer_dependency),
+            dataset_params=Depends(self.dataset_dependency),
+            post_process=Depends(self.process_dependency),
+            rescale=Depends(self.rescale_dependency),
+            color_formula=Depends(self.color_formula_dependency),
+            colormap=Depends(self.colormap_dependency),
+            render_params=Depends(self.render_dependency),
+            env=Depends(self.environment_dependency),
+        ):
+            """Create map tile from a dataset."""
+            tms = self.supported_tms.get(tileMatrixSetId)
+
+            reader_options = reader_params.as_dict()
+            if getattr(tile_params, "multiscale", False):
+                reader_options["group"] = z
+
+            with rasterio.Env(**env):
+                with self.reader(src_path, tms=tms, **reader_options) as src_dst:
+                    image = src_dst.tile(
+                        x,
+                        y,
+                        z,
+                        tilesize=scale * 256,
+                        **layer_params.as_dict(),
+                        **dataset_params.as_dict(),
+                    )
+
+            if post_process:
+                image = post_process(image)
+
+            if rescale:
+                image.rescale(rescale)
+
+            if color_formula:
+                image.apply_color_formula(color_formula)
+
+            content, media_type = render_image(
+                image,
+                output_format=format,
+                colormap=colormap,
+                **render_params.as_dict(),
+            )
+
+            return Response(content, media_type=media_type)
+
+    # custom /statistics endpoints (remove /statistics - GET)
+    def statistics(self):
+        """add statistics endpoints."""
+
+        # POST endpoint
+        @self.router.post(
+            "/statistics",
+            response_model=StatisticsGeoJSON,
+            response_model_exclude_none=True,
+            response_class=GeoJSONResponse,
+            responses={
+                200: {
+                    "content": {"application/geo+json": {}},
+                    "description": "Return dataset's statistics from feature or featureCollection.",
+                }
+            },
+        )
+        def geojson_statistics(
+            geojson: Annotated[
+                Union[FeatureCollection, Feature],
+                Body(description="GeoJSON Feature or FeatureCollection."),
+            ],
```

Comment:
NOTE: we should maybe extract this and make a dependency that user could customize to control the size of the GeoJSON